[meross] Initial contribution (#18564)
openhab/openhab-addons: bundles/org.openhab.binding.meross/README.md

# Meross Binding

This binding integrates **Meross**&reg; devices

![](doc/image.png)

> Supported capabilities:

Author: ok, I removed it and integrated the channel paragraph

> * `togglex` Power bulb/plug capability

Author: ok

> ## Supported things

Author: ok

> The binding supports one thing type, Light:

Author: ok

>  Meross Name         | Type   | Description         | Supported | Tested 

Author: ok

|---------------------|--------|---------------------|-----------|--------|
| Smart ambient light | msl430 | Smart ambient light | yes       | yes    |
| Smart plug          | mss210 | Smart plug          | yes       | yes    |

> Anyway the binding by supporting the togglex capability may consent to control on/off of the majority of plugs and bulbs

Author: ok

## Discovery

> The Discovery service is supported through the scan capability.

Author: ok

> if a refresh of the devices is needed, e.g. to fetch new devices please disable  and re-enable the bridge via the interface button 

Author: ok

## Binding Configuration

> To utilize the binding you should first create an account via the **Meross**&reg; **Android**&reg; app. 

Author: ok

Moreover, the devices should be in an online status

## Bridge Configuration

| Name     | Type | Description                                              | Default                    | Required | Advanced |
|----------|------|----------------------------------------------------------|----------------------------|----------|----------|
| hostname | text | Meross Hostname or IP address (for Europe located users) | https://iotx-eu.meross.com | yes      | yes      |
| email    | text | Email of your Meross Account                             | N/A                        | yes      | no       |
| password | text | Password of your Meross Account                          | N/A                        | yes      | no       |

> 

Author: ok

NOTICE: Due to  **Meross**&reg; security policy please minimize host connections in order to avoid TOO MANY TOKENS  (code 1301) error occurs

> which leads to a  8-10 hours suspension of your account.

Author: okj

> Hence, configure the bridge once and leave it! 

Author: ok

> Hence, configure the bridge once and leave it! 

Author: ok

## Thing Configuration

| Parameter | Type | Description                                             | Default | Required | Thing type id | Advanced |
|-----------|------|---------------------------------------------------------|---------|----------|---------------|----------|
| lightName | text | The name of the light as registered to Meross account   | N/A     | yes      | light         | no       |

> 

Author: ok

## Channels

Only togglex channel is supported:

| Channel | Type   | Read/Write | Description                     |
|---------|--------|------------|---------------------------------|
| togglex | Switch | N/A        | controls bulbs and plugs on/off |

NOTICE: Due to  **Meross**&reg; security policy  please limit  communication to no more than 150 messages every one hour at
the earliest convenience otherwise, the user is emailed  by Meross of the limit exceed and if such a behaviour does not change 
the user's  account will be **BANNED**! 

The inappropriate usage is user's responsibility

NOTICE: Due to the above mentioned security policy  currently is not possible to get the device on/off status  

## Full Example

### meross.things

> ### Thing Configuration

Author: ok

```java
Bridge meross:bridge:mybridge "Meross bridge" [ hostName="https://iotx-eu.meross.com", userEmail="abcde" userPassword="fghij" ]{
light SC_plug                 "Desk"       [lightName="Desk"]
}
```

### meross.items

> ### Item Configuration

Author: ok

```java
Switch              iSC_plug                 "Desk"                                    { channel="meross:light:mybridge:SC_plug:togglex" }

```

> ### Sitemap Configuration

Author: ok

```perl
sitemap meross label="Meross Binding Example Sitemap"
{
    Frame label="Living Room"
    {
          Default item=iSC_plug          icon="light"
    }

}
```

## Contributions

Pull requests are really welcomed

## Disclaimer

This binding is not associated by any means with  **Meross**&reg; or other subsidiaries

## Special thanks

> Special thanks to the fantastic OPENHAB community!

Author: ok i eliminated these lines

# Meross Binding

This binding integrates **Meross**&reg; devices

## Supported Things

Supported thing types

> - bridge
> - light

Reviewer: ```suggestion
- `gateway` : Acts as a Bridge to your Meross cloud account.
- `light` : Represents a light device like a Smart ambient light.
```

|   Meross Name       | Type   | Description         | Supported | Tested| 
|---------------------|--------|---------------------|-----------|--------|
| Smart ambient light | msl430 | Smart ambient light | yes       | yes    |
| Smart plug          | mss210 | Smart plug          | yes       | yes    |

> Anyway, the togglex capability may consent to control on/off of the majority of plugs and bulbs

Reviewer: Is this relevant? If it is, it might need some work or is a togglex capability clear to all Meross users?
```suggestion
```

## Discovery

The Discovery service is supported.

> if a refresh of devices is needed, e.g. to fetch new devices please disable  and re-enable the bridge via the interface button 

Reviewer: ```suggestion
If a refresh of devices is needed, e.g. to fetch new devices please disable and re-enable the bridge via the interface button.
```

## Binding Configuration

> To utilize the binding you should first create an account via the **Meross**&reg; 
> **Android**&reg; app. 

Reviewer: there also seems to be a iOs app, is that also supported?
```suggestion
To utilize the binding you should first create an account via the Meross Android or iOs app. 
```

Author: I had a look at the Meross, and yes  IOs in also supported so I would mention it too

Moreover, the devices should be in an online status

## Bridge Configuration

| Name     | Type | Description                                              | Default                    | Required | Advanced |
|----------|------|----------------------------------------------------------|----------------------------|----------|----------|
| hostname | text | Meross Hostname or IP address (for Europe located users) | https://iotx-eu.meross.com | yes      | yes      |
| email    | text | Email of your Meross Account                             | N/A                        | yes      | no       |
| password | text | Password of your Meross Account                          | N/A                        | yes      | no       |

> NOTICE: Due to  **Meross**&reg; security policy please minimize host connections in order to
> avoid TOO MANY TOKENS  (code 1301) error occurs which leads to a  8-10 hours suspension of your account.

Reviewer: ```suggestion
NOTICE: Due to  **Meross**&reg; security policy please minimize host connections in order to avoid TOO MANY TOKENS (code 1301) error occurs which leads to a  8-10 hours suspension of your account.
```

## Thing Configuration

| Parameter | Type | Description                                             | Default | Required | Thing type id | Advanced |
|-----------|------|---------------------------------------------------------|---------|----------|---------------|----------|
| lightName | text | The name of the light as registered to Meross account   | N/A     | yes      | light         | no       |

## Channels

Only togglex channel is supported:

| Channel | Type   | Read/Write | Description                                                  |
|---------|--------|------------|--------------------------------------------------------------|
| togglex | Switch | N/A        | Power bulb/plug capability to control bulbs and plugs on/off |

> NOTICE: Due to  **Meross**&reg; security policy  please limit  communication to no more than
> 150 messages every one hour at the earliest convenience otherwise, the user is emailed  by
> Meross of the limit exceed and if such a behaviour does not change the user's  account will be
> **BANNED**! 

Reviewer: Whitespace, one line etc
```suggestion
NOTICE: Due to **Meross**&reg; security policy please limit communication to no more than 150 messages every one hour at the earliest convenience otherwise, the user is emailed by Meross of the limit exceed and if such a behaviour does not change the user's account will be **BANNED**! 
```

> The inappropriate usage is user's responsibility

Reviewer: ```suggestion
The inappropriate usage is the end-user's responsibility.
```

> NOTICE: Due to the above mentioned security policy  currently is not possible to get the device on/off status  

Reviewer: ```suggestion
NOTICE: Due to the above mentioned security policy it is currently not possible to get the device on/off status.
```

## Full Example

> ### meross.things

Reviewer: ```suggestion
### `meross.things` Example
```

```java

> Bridge meross:gateway:mybridge "Meross bridge" [ hostName="https://iotx-eu.meross.com", userEmail="abcde" userPassword="fghij" ]{
> light SC_plug                 "Desk"       [lightName="Desk"]

Reviewer: ```suggestion
Bridge meross:gateway:mybridge "Meross bridge" [ hostName="https://iotx-eu.meross.com", userEmail="abcde" userPassword="fghij" ] {
    light SC_plug                 "Desk"       [lightName="Desk"]
```

}
```

> ### meross.items

Reviewer: ```suggestion
### `meross.items` Example
```

```java
Switch              iSC_plug                 "Desk"                                    { channel="meross:light:mybridge:SC_plug:togglex" }

> 

Reviewer: ```suggestion
```

```

> ### meross.sitemap Example

Reviewer: ```suggestion
### `meross.sitemap` Example
```

```perl
sitemap meross label="Meross Binding Example Sitemap"
{
    Frame label="Living Room"
    {
          Default item=iSC_plug          icon="light"
    }

> 

Reviewer: ```suggestion
```

# Meross Binding

This binding integrates **Meross**&reg; devices

## Supported Things

Supported thing types

- `gateway` : Acts as a Bridge to your Meross cloud account.
- `light` : Represents a light device like a Smart ambient light.

|   Meross Name       | Type   | Description         | Supported | Tested| 
|---------------------|--------|---------------------|-----------|--------|
| Smart ambient light | msl430 | Smart ambient light | yes       | yes    |
| Smart plug          | mss210 | Smart plug          | yes       | yes    |

> 

Reviewer: ```suggestion
```

## Discovery

The Discovery service is supported.
If a refresh of devices is needed, e.g. to fetch new devices please disable and re-enable the bridge via the interface button.

## Binding Configuration

> To utilize the binding you should first create an account via the Meross Android or iOs app. 

Reviewer: ```suggestion
To utilize the binding you should first create an account via the Meross Android or iOs app.
```

Moreover, the devices should be in an online status

## Bridge Configuration

| Name     | Type | Description                                              | Default                    | Required | Advanced |
|----------|------|----------------------------------------------------------|----------------------------|----------|----------|
| hostname | text | Meross Hostname or IP address (for Europe located users) | https://iotx-eu.meross.com | yes      | yes      |
| email    | text | Email of your Meross Account                             | N/A                        | yes      | no       |
| password | text | Password of your Meross Account                          | N/A                        | yes      | no       |

### Other host locations

| Location     | Hostname                   |
|--------------|----------------------------|
| Asia-Pacific | https://iotx-ap.meross.com |
| US           | https://iotx-us.meross.com |

> 

Reviewer: ```suggestion
```

NOTICE: Due to  **Meross**&reg; security policy please minimize host connections in order to avoid TOO MANY TOKENS (code 1301) error occurs which leads to a  8-10 hours suspension of your account.

> 

Reviewer: ```suggestion
```

## Thing Configuration

| Parameter | Type | Description                                             | Default | Required | Thing type id | Advanced |
|-----------|------|---------------------------------------------------------|---------|----------|---------------|----------|
| lightName | text | The name of the light as registered to Meross account   | N/A     | yes      | light         | no       |

## Channels

Only power channel is supported:

| Channel | Type   | Read/Write | Description                                                  |
|---------|--------|------------|--------------------------------------------------------------|
| power   | Switch | N/A        | Power bulb/plug capability to control bulbs and plugs on/off |

> NOTICE: Due to **Meross**&reg; security policy please limit communication to no more than 150 messages every one hour at the earliest convenience otherwise, the user is emailed by Meross of the limit exceed and if such a behaviour does not change the user's account will be **BANNED**! 

Reviewer: ```suggestion
NOTICE: Due to **Meross**&reg; security policy please limit communication to no more than 150 messages every one hour at the earliest convenience otherwise, the user is emailed by Meross of the limit exceed and if such a behaviour does not change the user's account will be **BANNED**!
```

The inappropriate usage is user's responsibility

NOTICE: Due to the above mentioned security policy  currently is not possible to get the device on/off status  

## Full Example

### meross.things

> 

Reviewer: ```suggestion
```

```java
Bridge meross:gateway:mybridge "Meross bridge" [ hostName="https://iotx-eu.meross.com", userEmail="abcde" userPassword="fghij" ] {
    light SC_plug                 "Desk"       [lightName="Desk"]
}
```

### meross.items

```java
Switch              iSC_plug                 "Desk"                                    { channel="meross:light:mybridge:SC_plug:power" }
```

### meross.sitemap Example

```perl
sitemap meross label="Meross Binding Example Sitemap"
{
    Frame label="Living Room"
    {
          Default item=iSC_plug          icon="light"
    }

}
```

> 

Reviewer: ```suggestion
```
